data/newtabicons-content.js

```diff
@@ -9,37 +9,38 @@
 'use strict';
 
 function mouseOverListener(e) {
-  let thumb = e.target.previousElementSibling;
+  if (e.target.previousElementSibling) {
+    var thumb = e.target.previousElementSibling;
+  } else {
+    var thumb = e.target;
+  }
   thumb.style.backgroundImage = thumb.dataset.newPreview;
 }
 
 function mouseOutListener(e) {
-  let thumb = e.target.previousElementSibling;
+  if (e.target.previousElementSibling) {
+    var thumb = e.target.previousElementSibling;
+  } else {
+    var thumb = e.target;
+  }
   thumb.style.backgroundImage = thumb.dataset.oldPreview;
 }
 
 function mouseMoveListener (e) {
-  var thumbnails = window.document.getElementsByClassName('newtab-cell');
-  for (let i = 0; i < thumbnails.length; ++i) {
-    let thumb = thumbnails[i];
-    let thumbs = thumb.getElementsByClassName('newtab-thumbnail');
-    if (!thumbs.length) {
-      thumbs = [thumb];
-    }
+  let thumbs = window.document.querySelectorAll('.newtab-thumbnail, .spotlight-image, .tile-img-container .site-icon-background');
```

Comment:
So, I think the complicated code there was to handle the cases where we had fewer thumbnails than cells (because we have a new profile, which doesn't have 12 suggested sites…)  You can see that in the bottom-right here:
<img width="1548" alt="screenshot 2016-06-24 09 52 52" src="https://cloud.githubusercontent.com/assets/159571/16339581/77883f4e-39f1-11e6-832d-038169b47f23.png">
Does this code still work with that, or do we need to do some post-processing?